### PR TITLE
Bump cert-manager to 1.12.0.

### PR DESCRIPTION
**Pull request by inelpandzic**
**CHANGE DESCRIPTION**
---
**Problem:**
Old cert-manager version. Bump it to 1.12.0.

**CHECKLIST**
---
**Jira**
- [ ] Is the Jira ticket created and referenced properly?
- [ ] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [ ] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PG version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Review by tplavcic — CHANGES_REQUESTED** [review]
Please bump the cert-manager version in e2e-tests/functions.

**Review by egegunes — APPROVED** [review]

File: e2e-tests/functions [hunk 1]
```diff
@@ -124,7 +124,7 @@ create_namespace() {
 deploy_cert_manager() {
 	kubectl_bin create namespace cert-manager || :
 	kubectl_bin label namespace cert-manager certmanager.k8s.io/disable-validation=true || :
-	kubectl_bin apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.9.0/cert-manager.yaml --validate=false || : 2>/dev/null
+	kubectl_bin apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.12.0/cert-manager.yaml --validate=false || : 2>/dev/null
```
Comment (hors) [inline e2e-tests/functions:127]: Can we use the latest v1.12.10?

**Review by hors — CHANGES_REQUESTED** [review]

File: e2e-tests/functions (hunk 1 above)
Comment (inelpandzic) [reply e2e-tests/functions:127]: I think yes

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| version-service | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-757/14c649e4c/version-service.log) |
| upgrade | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-757/14c649e4c/upgrade.log) |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-757/14c649e4c/init-deploy.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-757/14c649e4c/scheduled-backup.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-757/14c649e4c/demand-backup.log) |
| smart-update | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-757/14c649e4c/smart-update.log) |
| We run 6 out of 17|

commit: https://github.com/percona/percona-postgresql-operator/pull/757/commits/14c649e4c7e94c6e017415f5844b68e982ac25b5
image: `perconalab/percona-postgresql-operator:PR-757-14c649e4c-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-757-14c649e4c-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-757-14c649e4c-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-757-14c649e4c-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-757-14c649e4c-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-757-14c649e4c-pgo-deployer`

**Pull request closed without merging**